estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,83 @@
+from odoo import api, fields, models
+from datetime import timedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property offer Table"
+
+    price = fields.Float("Price")
+    status = fields.Selection(
+        [("accepted", "Accepted"), ("refused", "Refused")], string="Status", copy=False
+    )
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    validity = fields.Integer("Validity", default=7)
+    date_deadline = fields.Date(
+        compute="_date_deadline",
+        inverse="_inverse_date_deadline",
+        string="Date Deadline",
+    )
+    property_type_id = fields.Many2one(
+        "estate.property.type", related="property_id.property_type_id", store=True
+    )
+    _sql_constraints = [
+        (
+            "check_price",
+            "CHECK(price >= 0)",
+            "The offer price must be strictly positive",
+        )
+    ]
+    _order = "price desc"
+
+    @api.model
+    def create(self, vals):
+        property_id = vals.get("property_id")
```

Comment:
This wont work if multiple records are created. i.e. getting list of dict in vals

Reply:
We don't have the functionality to add multiple offers for one property at a time, so do we need to implement this in our module?